src/main/java/com/fauna/types/NamedDocument.java

```diff
@@ -5,54 +5,69 @@
 import java.util.Objects;
 
 /**
- * Represents a document that has a "name" instead of an "id". For example, a Role document is
- * represented as a NamedDocument.
+ * Represents a document identified by a "name" rather than an "id".
+ * This class is commonly used for documents where a unique name
+ * (e.g., for a Role) is more relevant than a numeric or autogenerated ID.
```

Comment:
```suggestion
 * Represents a document identified by a "name" rather than an "id".
 * This class is commonly used for documents in system collections where a unique name
 * (e.g., for a Role) is more relevant than a numeric or auto-generated ID.
```